openedx/features/subscriptions/models.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from datetime import date
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models import Q
+from model_utils.models import TimeStampedModel
+
+from student.models import CourseEnrollment
+
+class UserSubscription(TimeStampedModel):
+    """
+    The "UserSubscription" model.
+    """
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    subscription_id = models.PositiveIntegerField(db_index=True)
+    sku = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        verbose_name="SKU",
+    )
+    expiration_date = models.DateField(default=None, null=True, blank=True)
+
+    LIMITED_ACCESS = 'limited-access'
+    FULL_ACCESS_COURSES = 'full-access-courses'
+    FULL_ACCESS_TIME_PERIOD = 'full-access-time-period'
+    LIFETIME_ACCESS = 'lifetime-access'
+
+    SUBSCRIPTION_TYPE_CHOICES = (
+        (LIMITED_ACCESS, 'Limited Access', ),
+        (FULL_ACCESS_COURSES, 'Full Access (Courses)', ),
+        (FULL_ACCESS_TIME_PERIOD, 'Full Access (Time period)', ),
+        (LIFETIME_ACCESS, 'Lifetime Access', ),
+    )
+
+    subscription_type = models.CharField(choices=SUBSCRIPTION_TYPE_CHOICES, max_length=30)
+    number_of_courses = models.PositiveIntegerField(null=True, blank=True)
+    course_enrollments = models.ManyToManyField(CourseEnrollment, blank=True)
+
+
+    class Meta(object):
+        unique_together = ('subscription_id', 'user')
+
+    @classmethod
+    def get_active_subscription(self):
```

Comment:
we need to change the logic in this method get_active_subscription  . to determine if a user’s subscription is active or not
1. In case of (limited time and unlimited courses) subscription its expiry date should be greater than or equal to today.
2. In case of (unlimited time and limited courses) subscription it should have one or more courses enrollments available to be utilized
3. In case of (limited time and limited courses) its expiry date should be greater than or equal to today and it should have one or more course enrollments available to be utilized.
4. In case (unlimited time and unlimited courses) it is alway active.

Reply:
@ziafazal Done.